src/createGetDataElementValue.js

```diff
@@ -78,15 +74,15 @@ module.exports = function (
     }
 
     if (storageDuration) {
-      if (isDataElementValuePresent(value)) {
+      if (value != null) {
         dataElementSafe.setValue(name, storageDuration, value);
       } else {
         value = dataElementSafe.getValue(name, storageDuration);
       }
     }
 
-    if (!isDataElementValuePresent(value)) {
-      value = dataDef.defaultValue || '';
+    if (value == null && dataDef.defaultValue != null) {
```

Comment:
we're also double checking that `dataDef.defaultValue` is set so we don't accidentally change a semantic `null` in javascript to `undefined` by always falling back to `dataDef.defaultValue` (when value is null) and pulling out an undefined when `dataDef.defaultValue` is not actually set.

Reply:
I am not up to date with all the changes, but I expect Forge will not emit the `defaultValue` key when there isn't a default value provided. I guess the  user won't be able to put null or undefined by in Blacksmith.

Would it be better that instead of `dataDef.defaultValue != null` to say `typeof dataDef.defaultValue !== 'undefined`? Maybe it won't happen in real life (like I said at the beginning), but if for some chance Forge will emit `defaultValue: null` sometime, then you won't use the value emitted by Forge if you use `dataDef.defaultValue != null`.

Reply:
We'll confirm with the back end guys regarding if there's a use-case to emit an actual `null` for defaultValue. Aaron and I confirmed that `undefined` key/value pairs are stripped from JSON when serialized.

Reply:
Resolving confirmed with the Blacksmith team that we are not emitting `defaultValue` when it is `undefined` or `null`.